lib/pure/includes/osenv.nim

```diff
@@ -201,33 +93,14 @@ else:
     ## * `existsEnv proc <#existsEnv,string>`_
     ## * `delEnv proc <#delEnv,string>`_
     ## * `envPairs iterator <#envPairs.i>`_
-
-    # Note: by storing the string in the environment sequence,
-    # we guarantee that we don't free the memory before the program
-    # ends (this is needed for POSIX compliance). It is also needed so that
-    # the process itself may access its modified environment variables!
-    when nimvm:
-      discard "built into the compiler"
+    when defined(windows):
+      if key.len == 0 or '=' in key:
```

Comment:
This `'=' notin key` is a new requirement.

Reply:
= in key causes bug, see https://github.com/nim-lang/Nim/issues/18502 which was already fixed by #18050 

```nim
import os
putEnv("foo2=bar", "baz")
echo getEnv("foo2=bar", "def") # 1.4: prints bar=baz
echo getEnv("foo2", "def") # 1.4: prints bar=baz
```
after https://github.com/nim-lang/Nim/pull/18530, `putEnv("foo2=bar", "baz")` raises on linux|osx

after this PR, it also raises on windows, making code more platform and correct:

https://docs.microsoft.com/en-us/windows/win32/procthread/environment-variables
> The name of an environment variable cannot include an equal sign (=).

Reply:
Add it to the changelog please then.

Reply:
done